api/src/main/java/gov/cms/ab2d/api/controller/BulkDataAccessAPI.java

```diff
@@ -186,6 +183,12 @@
         }
     }
 
+    private String getUrlPath(Job job, String filePath) {
+        String requestURIString = ServletUriComponentsBuilder.fromCurrentRequestUri().toUriString();
```

Comment:
Use `ServletUriComponentsBuilder.fromCurrentRequestUri().replacePath` method. It's cleaner and more flexible.